lib/puppet/parser/functions/consul_sorted_json.rb

```diff
@@ -1,43 +1,91 @@
 require 'json'
 
-def sorted_json(obj)
-  case obj
+module ConsulSortedJson
+
+  module_function
+
+
+  def sorted_json(config_hash, pretty)
+    cleaned = convert_integers( config_hash )
+    sorted  = sort_keys( cleaned )
+
+    if pretty
+      JSON.pretty_generate( sorted )
```

Comment:
yeah, that's been a concern. That concern isn't related to this pull request though - at least I don't how that concern would change from this implementation to the current one (to_json).